File: packages/idb-distribute/src/dbName.ts
export const DB_NAME_REGEX = /^[A-Za-z0-9_-]{1,64}$/;
export const DB_NAME_ERROR = "Invalid dbName: must match ^[A-Za-z0-9_-]{1,64}$";

export function isValidDbName(dbName: string): boolean {
  return DB_NAME_REGEX.test(dbName);
}

export function assertValidDbName(dbName: string): void {
  if (!isValidDbName(dbName)) {
    throw new Error(DB_NAME_ERROR);
  }
}

File: apps/syncdb-server/lib/db_name.ml
let error_message = "invalid dbName: must match ^[A-Za-z0-9_-]{1,64}$"

let is_allowed_char = function
  | 'a' .. 'z' | 'A' .. 'Z' | '0' .. '9' | '_' | '-' -> true
  | _ -> false

let is_valid db_name =
  let len = String.length db_name in
  len > 0 && len <= 64 && String.for_all is_allowed_char db_name

let validate db_name = if is_valid db_name then Ok db_name else Error error_message
